src/collectdcvmfs/__init__.py

```diff
@@ -1,4 +1,90 @@
-# Example package with a console entry point
+import collectd
+import xattr
+import os
+import psutil
+import time
+
+
+CVMFS_ROOT = '/cvmfs'
+PLUGIN_NAME = 'cvmfs'
+
+
+def read_mounttime(repo_mountpoint):
+    start = time.time()
+    os.listdir(repo_mountpoint)
```

Comment:
No timeout? Does collectd itself have a global one?

Reply:
If the probing doesn't finish in the interval, it's killed.

Does it sound like a reasonable behaviour ?

Plan B is to thread it and kill it, but I don't know what would be a reasonable timeout in this case.

autofs should be playing a role here as well, so depending how it's configured it could affect this in some cases.

Reply:
We will be safe from hanging for ever which is the main thing. Suggest we go and  if we find we need threads then maybe so. From memory you don't normally see cms.cern.ch hanging and atlas.cern.ch not hanging. Such a situation would justify threads but we can check once we have data.